game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -639,7 +639,7 @@ private void removeFromNonCombatLandings(
       if (landedTerritory == null) {
         throw new IllegalStateException("not unloaded?:" + units);
       }
-      remove(lost, bridge, landedTerritory, false);
+      remove(new ArrayList<>(lost), bridge, landedTerritory, false);
```

Comment:
I'm not sure if this is the best fix. Creating a copy probably should be the responsibility of the method, not the caller. Usually we would want APIs to be impossible to use and not have arcane contracts where just by using the API you break it.

It's also a bit odd as it's difficult to see what the side effect would be from either the parameters or the method name (yikes, the code here really could have used some improvements..)

What's the cost to moving the copy to the method? Failing that, perhaps it should be commented here why we create a copy, but that leaves unresolved that we have a *very* brittle API that can break just by using it in a very reasonable way.

Reply:
Done. I made a helper function which is also reused from another remove() implementation that avoids the extra copy if there are no dependent units.